components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java

```diff
@@ -428,14 +429,35 @@ private void persistAccessTokenInDB(OAuthTokenReqMessageContext tokReqMsgCtx, Ac
         }
     }
 
-    private void updateCacheIfEnabled(AccessTokenDO newTokenBean, String scope)
+    private void updateCacheIfEnabled(AccessTokenDO newTokenBean, String scope, OauthTokenIssuer oauthTokenIssuer)
             throws IdentityOAuth2Exception {
 
         if (isHashDisabled && cacheEnabled) {
-            OAuthCacheKey cacheKey = getOAuthCacheKey(scope, newTokenBean.getConsumerKey(),
-                    newTokenBean.getAuthzUser().toString(), newTokenBean.getAuthzUser().getFederatedIdPName(),
-                    getTokenBindingReference(newTokenBean));
-            oauthCache.addToCache(cacheKey, newTokenBean);
+            AccessTokenDO tokenToCache = AccessTokenDO.clone(newTokenBean);
+            // If usePersistedAccessTokenAlias is enabled then in the DB the
+            // access token alias taken from the OauthTokenIssuer's getAccessTokenHash
+            // method is set as the token.
+            if (oauthTokenIssuer.usePersistedAccessTokenAlias()) {
+                try {
+                    String jti = oauthTokenIssuer.getAccessTokenHash(newTokenBean.getAccessToken());
+                    tokenToCache.setAccessToken(jti);
```

Comment:
it would better to change the jti variable name to something else :)
persistedAlias or something

Reply:
Resolved with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1342 and https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1343